### PR TITLE
FEAT: ORV2-1865 Implementing Feature Flags

### PR DESCRIPTION
**Pull request by erikataot**
# Description
- Implementing Feature Flags in the client and the API
- Should be available in the client after the user authenticates themselves

Screenshot 2024-02-22 121900

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1214-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1214-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1214-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1214-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1214-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1214-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1214-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1214-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

**Review by zgong-gov — APPROVED** [review]

File: dops/src/modules/feature-flags/profiles/feature-flags.profile.ts
Comment (praju-aot) [inline dops/src/modules/feature-flags/profiles/feature-flags.profile.ts:1]: Profile not needed for dops and TPS as entity need not be mapped to DTO

File: dops/src/modules/feature-flags/dto/response/read-feature-flag.dto.ts
Comment (praju-aot) [inline dops/src/modules/feature-flags/dto/response/read-feature-flag.dto.ts:1]: DTO not needed for dops and TPS as the object is not being sent across the network

File: tps-migration/src/modules/feature-flags/dto/response/read-feature-flag.dto.ts
Comment (praju-aot) [inline tps-migration/src/modules/feature-flags/dto/response/read-feature-flag.dto.ts:1]: DTO not needed for dops and TPS as the object is not being sent across the network

File: tps-migration/src/modules/feature-flags/profiles/feature-flags.profile.ts
Comment (praju-aot) [inline tps-migration/src/modules/feature-flags/profiles/feature-flags.profile.ts:1]: Profile not needed for dops and TPS as entity need not be mapped to DTO

**Review by praju-aot — CHANGES_REQUESTED** [review]

**Review by zgong-gov — COMMENTED** [review]
Approved for FE

**Review by praju-aot — APPROVED** [review]

**Review by krishnan-aot — COMMENTED** [review]
image
 
I'm seeing a lot of unnecessary calls just on the login screen. Can you make it so that the call to load feature flag happens only after login?

**Pull request closed without merging**